### ISSUE TITLE
Compatability of ExceptionHandler with non-static classes?

### ISSUE THREAD

**Issue by mcquiggd** [Jusas/AzureFunctionsV2.HttpExtensions#1]
This looks a very interesting project...

I noticed that the Fody weaving for ExceptionHandling is targeting only static methods in static classes - was this a decision made for a specific reason?

I am using constructor injection for dependencies, so I require non-static classes...

**Comment by Jusas** [comment]
Yep, this is currently a limitation, I'll have to take a look at it since DI is now pretty much the norm.
